core/src/main/scala-3/sttp/tapir/macros/TapirMacros.scala

```diff
@@ -17,8 +20,33 @@ trait TapirMacros {
     *
     * Should be used in [[oneOf]] output descriptions.
     */
-  def oneOfMapping[T: ClassTag](statusCode: StatusCode, output: EndpointOutput[T]): OneOfMapping[T] = ??? // TODO
+  inline def oneOfMapping[T: ClassTag](statusCode: StatusCode, output: EndpointOutput[T]): OneOfMapping[T] = 
+    ${ TapirMacros.oneOfMappingImpl[T]('statusCode, 'output, '{classTag[T]}) }
 
   @scala.deprecated("Use oneOfMapping", since = "0.18")
-  def statusMapping[T: ClassTag](statusCode: StatusCode, output: EndpointOutput[T]): OneOfMapping[T] = ??? // TODO
+  inline def statusMapping[T: ClassTag](statusCode: StatusCode, output: EndpointOutput[T]): OneOfMapping[T] = 
+    ${ TapirMacros.oneOfMappingImpl[T]('statusCode, 'output, '{classTag[T]}) }
+
 }
+
+object TapirMacros {
+  def oneOfMappingImpl[T: Type](statusCode: Expr[StatusCode], output: Expr[EndpointOutput[T]], ct: Expr[ClassTag[T]])(using Quotes): Expr[OneOfMapping[T]] = {
+    import quotes.reflect._
+
+    val t = TypeRepr.of[T]
+
+    val isAllowed: TypeRepr => Boolean = {
```

Comment:
maybe add a comment that this should check the erasure, with a link to the GH dotty issue?